test/object-store/util/sync/sync_test_utils.hpp

```diff
@@ -47,6 +47,26 @@
 #define REALM_ENABLE_AUTH_TESTS 0
 #endif
 
+#if REALM_ENABLE_SYNC
+CATCH_REGISTER_ENUM(realm::SyncClientHookEvent, realm::SyncClientHookEvent::DownloadMessageReceived,
+                    realm::SyncClientHookEvent::DownloadMessageIntegrated,
+                    realm::SyncClientHookEvent::BootstrapMessageProcessed,
+                    realm::SyncClientHookEvent::BootstrapProcessed, realm::SyncClientHookEvent::ErrorMessageReceived,
+                    realm::SyncClientHookEvent::SessionActivating, realm::SyncClientHookEvent::SessionSuspended,
+                    realm::SyncClientHookEvent::SessionConnected, realm::SyncClientHookEvent::SessionResumed,
+                    realm::SyncClientHookEvent::BindMessageSent, realm::SyncClientHookEvent::IdentMessageSent,
+                    realm::SyncClientHookEvent::ClientResetMergeComplete,
+                    realm::SyncClientHookEvent::BootstrapBatchAboutToProcess,
+                    realm::SyncClientHookEvent::UploadMessageSent)
+
+namespace realm {
+inline std::ostream& operator<<(std::ostream& os, const realm::SyncClientHookEvent& event)
```

Comment:
Do we want this in the realm namespace?

Reply:
I'm not 100% sure why, but the compiler was not able to find it in the global namespace - I believe it's because the `operator<<` function needs to be in one of the same namespaces as its parameters (e.g. `std` or `realm`).

Reply:
i see. not sure it's a good idea though. Jonathan seemed to suggest it's not needed.

Reply:
removed - but left in the `CATCH_REGISTER_ENUM` definition